plugin/resurrect/fuzzy_loader.lua

```diff
@@ -15,35 +27,391 @@ function pub.get_default_fuzzy_load_opts()
+	else
+		label.name_fmt = label.name_raw
+		label.name_len = utf8len(label.name_fmt)
+	end
+
+	-- check the overall width against the available width
+	local width = label.name_len + label.date_len + 4
+	-- `overflow_chars` is the number of character we should remove to fit within the size of the window
+	local overflow_chars = math.max(0, width - win_width)
+
+	if overflow_chars == 0 then
+		-- No oversize for this line, thus we keep it as is
+		return label.name_fmt .. label.date_fmt
+	else
+		-- we need to remove characters; first check if the padding can be used
+		if label.padding_len ~= 0 then
+			local new_len = math.max(0, label.padding_len - overflow_chars)
+			overflow_chars = overflow_chars - (label.padding_len - new_len) -- update the oversize
+			label.padding_raw = string.rep(".", new_len)
+		end
+		-- we tackle the filename reducing it to a length with a minimum of `min_filename_len`
+		if overflow_chars ~= 0 then
+			-- new we need to apply the size reduction to the filename, our strategy:
+			-- remove the `oversize` from the middle of the filename string and
+			-- replace it by opts.name_truncature, thus we need to correct that by adding its length
+			overflow_chars = overflow_chars + pad_len
+			-- here we can re-adjust the filename string to fit the available room, but up to a point
+			local reduction = label.filename_len - math.max(min_filename_len, label.filename_len - overflow_chars)
+			overflow_chars = overflow_chars - reduction
+			label.filename_raw = utils.replace_center(label.filename_raw, reduction, str_pad)
+		end
+		-- do we still have an oversize? we can do something only if we have a date, otherwise we did our best
+		if overflow_chars ~= 0 and opts.show_state_with_date then
+			local new_len = math.max(0, label.date_len - overflow_chars)
+			if new_len == 0 then
+				label.date_raw = ""
+			else
+				label.date_raw = label.date_raw:sub(1, new_len)
+			end
+		end
+	end
+
+	-- now we can format and recombine the reduced strings
+	result = label.filename_raw .. label.separator .. label.padding_raw .. label.separator
+	if file.fmt then
+		result = file.fmt(result)
+	end
+	if opts.show_state_with_date then
+		if opts.fmt_date then
+			result = result .. opts.fmt_date(label.date_raw)
+		else
+			result = result .. label.date_raw
+		end
+	end
+
+	return result
+end
+
+-- build a table with the output of the file finder function
+---@param stdout string|nil
+---@param opts table
+---@return table
+local function insert_choices(stdout, opts)
+	-- pre-calculation of formatting cost
+	local types = { "workspace", "window", "tab" }
+	local state_files = {}
+	local files = {}
+	local max_length = 0
+
+	if stdout == nil then
+		return state_files
+	end
+
+	-- Parse the stdout and construct the file table
+	for line in stdout:gmatch("[^\n]+") do
+		local epoch, type, file = line:match("%s*(%d+)%s+.+[/\\]([^/\\]+)[/\\]([^/\\]+%.json)$")
+		if epoch and file and type and type == type then
```

Comment:
unnecessary check on itself
```suggestion
		if epoch and file and type then
```